### PR TITLE
Patch for feature/improved robot state publisher

### PR DESCRIPTION
**Pull request by lrapetti**
Addressing the following change requests in https://github.com/robotology/idyntree/pull/513:
- https://github.com/robotology/idyntree/pull/513#discussion_r249465421
- https://github.com/robotology/idyntree/pull/513#pullrequestreview-201211455

### CODE REVIEW

File: src/tools/yarprobotstatepublisher/include/robotstatepublisher.h [hunk 1]
```diff
@@ -27,13 +27,13 @@
...
 /****************************************************************/
-class JointStateSuscriber: public yarp::os::Subscriber<yarp::rosmsg::sensor_msgs::JointState>
+class JointStateSubcriber: public yarp::os::Subscriber<yarp::rosmsg::sensor_msgs::JointState>
 {
 private:
     YARPRobotStatePublisherModule* m_module;
 
 public:
-    JointStateSuscriber();
+    JointStateSubcriber();
```
Comment (prashanthr05) [inline src/tools/yarprobotstatepublisher/include/robotstatepublisher.h]: Subcriber -> Subscriber

File: src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp [hunk 2]
```diff
@@ -142,7 +142,7 @@ bool YARPRobotStatePublisherModule::configure(ResourceFinder &rf)
 
     // Setup the topic and configureisValid the onRead callback
     string jointStatesTopicName = rf.check("jointstates-topic",Value("/joint_states")).asString();
-    m_jointStateSubscriber = new JointStateSuscriber();
+    m_jointStateSubscriber = new JointStateSubcriber();
```
Comment (prashanthr05) [inline src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp]: `new` is a bad thing to use. As always, its better to replace it with smart pointers, if possible.

File: src/tools/yarprobotstatepublisher/include/robotstatepublisher.h (hunk 1 above)
Comment (lrapetti) [reply src/tools/yarprobotstatepublisher/include/robotstatepublisher.h]: done https://github.com/robotology/idyntree/pull/535/commits/d191bc9b2f41d991844cc64e0809d5634e882965

File: src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp (hunk 2 above)
Comment (lrapetti) [reply src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp]: https://github.com/robotology/idyntree/pull/535/commits/a4611e241fdf6704e0bdf125ee5e079858a40383

**Comment by traversaro** [comment]
The travis build are failing because all those branches are out of date w.r.t. to master. 
To proceeed in a clean way, I suggest that you just create a new branch in your local fork and open a new PR from that, thanks @lrapetti !

**Comment by traversaro** [comment]
Superseded by https://github.com/robotology/idyntree/pull/538.

**Pull request closed without merging**